x-pack/elastic-agent/pkg/agent/configuration/instrumentation.go

```diff
@@ -0,0 +1,20 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package configuration
+
+// InstrumentationConfig configures APM Tracing.
+type InstrumentationConfig struct {
```

Comment:
I think we need to add tls config here as well

Reply:
That's a good point. Internally, `apmtransport.NewHTTPTransport()` is checking environment variables to find TLS configs. While that works, it is leaving TLS config to a different method (environment variables vs. explicitly setting in the yml config).

Based on [this comment](https://github.com/elastic/fleet-server/pull/898#discussion_r753943397) in the fleet instrumentation PR, it sounds like they're not wanting to handle environment variables.

@marclop do you have an idea how you'd like to handle TLS configuration? I'm not sure if you considered it for the PR, since it's just a first iteration.

Reply:
I did not address that in my initial PR and I had overlooked that part. I'm happy to open a follow up with what we decide here.